atomic_reactor/yum_util.py

```diff
@@ -0,0 +1,112 @@
+"""
+Copyright (c) 2018 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+Utility functions to manipulate a yum repo. Guarantees that a yum repo will have a unique
+name and the .repo suffix.
+
+"""
+from atomic_reactor.constants import YUM_REPOS_DIR
+from atomic_reactor.util import get_retrying_requests_session
+from hashlib import md5
+import os
+import os.path
+import logging
+from copy import deepcopy
+
+try:
+    # py2
+    from urlparse import unquote, urlsplit
+    import ConfigParser as configparser
+    # We import BytesIO as StringIO as configparser can't properly write
+    from io import BytesIO, BytesIO as StringIO
+    YUM_TEST_STRING = 'line noise'
+    print("python 3.6 what?")
+except ImportError:
+    # py3
+    from urllib.parse import unquote, urlsplit
+    import configparser
+    from io import BytesIO, StringIO
+    YUM_TEST_STRING = u'line noise'
+    print("python 3.6 is good to go")
+
+
+logger = logging.getLogger(__name__)
+
+REPO_SUFFIX = ".repo"
+
+
+class YumRepo(object):
+    def __init__(self, repourl=None, content=None, dst_repos_dir=YUM_REPOS_DIR):
```

Comment:
why are 'repourl' isn't mandatory like it was before?

Reply:
Fixed.